ibm/resource_ibm_cis_global_load_balancer.go

```diff
@@ -0,0 +1,227 @@
+			},
+			"default_pool_ids": {
+				Type:     schema.TypeSet,
+				Required: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				//ValidateFunc: validation.StringLenBetween(1, 32),
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+				//ValidateFunc: validation.StringLenBetween(0, 1024),
+			},
+			"ttl": {
+				Type:          schema.TypeInt,
+				Optional:      true,
+				Computed:      true,
+				ConflictsWith: []string{"proxied"}, // this is set to zero regardless of config when proxied=true
+
+			},
+			"proxied": {
+				Type:          schema.TypeBool,
+				Optional:      true,
+				Default:       false,
+				ConflictsWith: []string{"ttl"},
+			},
+			"session_affinity": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "none",
+				// Set to cookie when proxy=true
+				ValidateFunc: validateAllowedStringValue([]string{"none", "cookie"}),
+			},
+			// "region_pools": &schema.Schema{
+			// 	Type:     schema.TypeMap,
+			// 	Optional: true,
+			// 	Computed: true,
+			// 	Elem:     &schema.Schema{Type: schema.TypeString},
+			// },
+			// "pop_pools": &schema.Schema{
+			// 	Type:     schema.TypeMap,
+			// 	Optional: true,
+			// 	Computed: true,
+			// 	Elem:     &schema.Schema{Type: schema.TypeString},
+			// },
+			"created_on": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"modified_on": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+
+		Create: resourceCISGlbCreate,
+		Read:   resourceCISGlbRead,
+		Update: resourceCISGlbUpdate,
+		Delete: resourceCISGlbDelete,
+		// No Exists due to errors in CIS API returning incorrect return codes on 404
+	}
+}
+
+func resourceCISGlbCreate(d *schema.ResourceData, meta interface{}) error {
+	cisClient, err := meta.(ClientSession).CisAPI()
+	log.Printf("   client %v\n", cisClient)
+	if err != nil {
+		return err
+	}
+
+	cisId := d.Get("cis_id").(string)
+	name := d.Get("name").(string)
+	zoneId := d.Get("domain_id").(string)
+	defaultPools := d.Get("default_pool_ids").(*schema.Set).List()
+
+	var Glbs *[]v1.Glb
+	Glbs, err = cisClient.Glbs().ListGlbs(cisId, zoneId)
+	if err != nil {
+		log.Printf("ListGlbs Failed %s\n", err)
+		return err
+	}
+
+	var GlbNames []string
+	GlbsObj := *Glbs
+	for _, Glb := range GlbsObj {
+		GlbNames = append(GlbNames, Glb.Name)
+	}
+
+	log.Println(string("Existing Glb names"))
+	log.Println(GlbNames)
+
+	var Glb *v1.Glb
+	var GlbObj v1.Glb
+
+	index := indexOf(name, GlbNames)
+	if index == -1 {
+
+		GlbNew := v1.GlbBody{}
+		GlbNew.Name = name
+
+		GlbNew.DefaultPools = expandToStringList(defaultPools)
+		GlbNew.FallbackPool = d.Get("fallback_pool_id").(string)
+		GlbNew.Proxied = d.Get("proxied").(bool)
+		GlbNew.SessionAffinity = d.Get("session_affinity").(string)
+
+		if description, ok := d.GetOk("description"); ok {
+			GlbNew.Desc = description.(string)
+		}
+		Glb, err = cisClient.Glbs().CreateGlb(cisId, zoneId, GlbNew)
```

Comment:
I'm not familiar with API of CIS...Instead of checking if glb existing with that name..when you try to call CreateGlb with params...doesn't the API itself through the error saying Already we have glb with same name please use different name

Reply:
Agreed n not necessary. Removed every occurance.